docs/platformsh.md

```diff
@@ -1,53 +0,0 @@
-## Platform.sh
-
-We use [platform.sh](http://platform.sh) for deploying Pull Requests to environments.
```

Comment:
I wouldn't get rid of these docs just yet, these might be helpful to help us debug issues with the environment

Reply:
@pvhee plat.sh has already a good documentation why do we need this? also they keep it up to date

I want to keep only relevant documentation

Reply:
@Saphyel ok, but let's then at least link off to plat.sh docs, and explain why we need / use feature branches

Reply:
That maybe part of CONTRIBUTING?